web/api.go

```diff
@@ -82,20 +84,108 @@ func handlerAPIQuotesSubmit(w http.ResponseWriter, r *http.Request) {
 
 	// Store UnverifiedQuote in database
 	err = database.CreateUnverifiedQuote(quote)
+
+	/* ------------------------- MISSING ERROR HANDLING ------------------------- */
+	// if err != nil {
+	// 	if strings.Contains(err.Error(), `violates foreign key constraint`) {
+	// 		w.WriteHeader(http.StatusBadRequest)
+	// 		fmt.Fprintf(w, "Teacher: no teacher with that ID")
+	// 	} else {
+	// 		w.WriteHeader(http.StatusInternalServerError)
+	// 		fmt.Fprintf(w, "internal server error")
+	// 		log.Printf("/api/quotes/submit: quote creation failed with error '%v' for request body '%s' and UnverifiedQuoteT %v", err, bytes, quote)
+	// 	}
+	// }
+}
+
+func handlerAPIUnverifiedQuotes(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(mux.Vars(r)["id"])
 	if err != nil {
-		if strings.Contains(err.Error(), `violates foreign key constraint "unverifiedquotes_teacherid_fkey"`) {
+		// This should not happend, because handlerAPIUnverifiedQuotes is only called if
+		// uri pattern is matched, see web.go
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "internal server error")
+		return
+	}
+	if id == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "invalid UnverifiedQuoteID: 0")
+		return
+	}
+
+	var subm quoteInputT
+	var quote database.UnverifiedQuoteT
+
+	switch r.Method {
+	case "PUT":
+
+		// parse json request body into temporary QuoteInput
+		bytes, _ := ioutil.ReadAll(r.Body)
+		err := json.Unmarshal(bytes, &subm)
+
+		if err != nil {
 			w.WriteHeader(http.StatusBadRequest)
-			fmt.Fprintf(w, "Teacher: no teacher with that ID")
-		} else {
-			w.WriteHeader(http.StatusInternalServerError)
-			fmt.Fprintf(w, "internal server error")
-			log.Printf("/api/quotes/submit: quote creation failed with error '%v' for request body '%s' and UnverifiedQuoteT %v", err, bytes, quote)
+			fmt.Fprintf(w, "unparsable JSON")
+			return
 		}
-	}
-}
 
-func handlerAPIUnverifiedQuotes(w http.ResponseWriter, r *http.Request) {
+		// Check validity of temporary QuoteInput and
+		// copy content into UnverifiedQuote
+
+		switch subm.Teacher.(type) {
+		case float64:
+			if subm.Teacher.(float64) == 0 {
+				w.WriteHeader(http.StatusBadRequest)
+				fmt.Fprintf(w, "invalid TeacherID: 0")
+				return
+			}
+			quote.TeacherID = int32(subm.Teacher.(float64))
+			quote.TeacherName = ""
+		case string:
```

Comment:
String needs a minimum length to be considered valid
... character ranges for all fields

Reply:
what i mean is that the following needs to be the case: ```subm.Teacher != ""```